File: components/autofill_assistant/browser/android/assistant_qr_code_camera_scan_model_wrapper.cc
// Copyright 2022 The Chromium Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

#include "components/autofill_assistant/browser/android/assistant_qr_code_camera_scan_model_wrapper.h"

#include "base/android/jni_string.h"
#include "components/autofill_assistant/android/jni_headers/AssistantQrCodeCameraScanModelWrapper_jni.h"

using base::android::AttachCurrentThread;
using base::android::ConvertUTF8ToJavaString;

namespace autofill_assistant {

AssistantQrCodeCameraScanModelWrapper::AssistantQrCodeCameraScanModelWrapper() {
  jni_env_ = AttachCurrentThread();
  java_assistant_camera_scan_model_wrapper_ =
      Java_AssistantQrCodeCameraScanModelWrapper_Constructor(jni_env_);
}

AssistantQrCodeCameraScanModelWrapper::
    ~AssistantQrCodeCameraScanModelWrapper() = default;

base::android::ScopedJavaGlobalRef<jobject>
AssistantQrCodeCameraScanModelWrapper::GetModel() const {
  return java_assistant_camera_scan_model_wrapper_;
}

void AssistantQrCodeCameraScanModelWrapper::SetDelegate(
    const base::android::ScopedJavaGlobalRef<jobject>&
        java_qr_code_native_delegate) const {
  Java_AssistantQrCodeCameraScanModelWrapper_setDelegate(
      jni_env_, java_assistant_camera_scan_model_wrapper_,
      java_qr_code_native_delegate);
}

void AssistantQrCodeCameraScanModelWrapper::SetToolbarTitle(
    const std::string& title_text) const {
  Java_AssistantQrCodeCameraScanModelWrapper_setToolbarTitle(
      jni_env_, java_assistant_camera_scan_model_wrapper_,
      ConvertUTF8ToJavaString(jni_env_, title_text));
}

void AssistantQrCodeCameraScanModelWrapper::SetPermissionText(
    const std::string& permission_text) const {
  Java_AssistantQrCodeCameraScanModelWrapper_setPermissionText(
      jni_env_, java_assistant_camera_scan_model_wrapper_,
      ConvertUTF8ToJavaString(jni_env_, permission_text));
}

void AssistantQrCodeCameraScanModelWrapper::SetPermissionButtonText(
    const std::string& permission_button_text) const {
  Java_AssistantQrCodeCameraScanModelWrapper_setPermissionButtonText(
      jni_env_, java_assistant_camera_scan_model_wrapper_,
      ConvertUTF8ToJavaString(jni_env_, permission_button_text));
}

void AssistantQrCodeCameraScanModelWrapper::SetOpenSettingsText(
    const std::string& open_settings_text) const {
  Java_AssistantQrCodeCameraScanModelWrapper_setOpenSettingsText(
      jni_env_, java_assistant_camera_scan_model_wrapper_,
      ConvertUTF8ToJavaString(jni_env_, open_settings_text));
}

void AssistantQrCodeCameraScanModelWrapper::SetOpenSettingsButtonText(
    const std::string& open_settings_button_text) const {
  Java_AssistantQrCodeCameraScanModelWrapper_setOpenSettingsButtonText(
      jni_env_, java_assistant_camera_scan_model_wrapper_,
      ConvertUTF8ToJavaString(jni_env_, open_settings_button_text));
}

void AssistantQrCodeCameraScanModelWrapper::SetCameraPreviewInstructionText(
    const std::string& camera_preview_instruction_text) const {
  Java_AssistantQrCodeCameraScanModelWrapper_setOverlayInstructionText(
      jni_env_, java_assistant_camera_scan_model_wrapper_,
      ConvertUTF8ToJavaString(jni_env_, camera_preview_instruction_text));
}

void AssistantQrCodeCameraScanModelWrapper::SetCameraPreviewSecurityText(
    const std::string& camera_preview_security_text) const {
  Java_AssistantQrCodeCameraScanModelWrapper_setOverlaySecurityText(
      jni_env_, java_assistant_camera_scan_model_wrapper_,
      ConvertUTF8ToJavaString(jni_env_, camera_preview_security_text));
}
}  // namespace autofill_assistant

File: components/autofill_assistant/android/java/src/org/chromium/components/autofill_assistant/qr_code/AssistantQrCodeCameraScanModelWrapper.java
// Copyright 2022 The Chromium Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

package org.chromium.components.autofill_assistant.qr_code;

import org.chromium.base.annotations.CalledByNative;
import org.chromium.base.annotations.JNINamespace;
import org.chromium.components.autofill_assistant.guided_browsing.qr_code.camera_scan.AssistantQrCodeCameraScanModel;

/**
 * Wrapper around the |AssistantQrCodeCameraScanModel| to manage the state for the QR Code Camera
 * Scan UI from the native code.
 */
@JNINamespace("autofill_assistant")
public class AssistantQrCodeCameraScanModelWrapper {
    private final AssistantQrCodeCameraScanModel mCameraScanModel;

    /**
     * The AssistantQrCodeCameraScanWrapperModel constructor.
     */
    @CalledByNative
    public AssistantQrCodeCameraScanModelWrapper() {
        mCameraScanModel = new AssistantQrCodeCameraScanModel();
    }

    /**
     * Returns the underlying AssistantQrCodeCameraScanModel object.
     */
    AssistantQrCodeCameraScanModel getCameraScanModel() {
        return mCameraScanModel;
    }

    @CalledByNative
    private void setDelegate(AssistantQrCodeNativeDelegate delegate) {
        mCameraScanModel.setDelegate(delegate);
    }

    @CalledByNative
    private void setToolbarTitle(String text) {
        mCameraScanModel.setToolbarTitle(text);
    }

    @CalledByNative
    private void setPermissionText(String text) {
        mCameraScanModel.setPermissionText(text);
    }

    @CalledByNative
    private void setPermissionButtonText(String text) {
        mCameraScanModel.setPermissionButtonText(text);
    }

    @CalledByNative
    private void setOpenSettingsText(String text) {
        mCameraScanModel.setOpenSettingsText(text);
    }

    @CalledByNative
    private void setOpenSettingsButtonText(String text) {
        mCameraScanModel.setOpenSettingsButtonText(text);
    }

    @CalledByNative
    private void setOverlayInstructionText(String text) {
        mCameraScanModel.setOverlayInstructionText(text);
    }

    @CalledByNative
    private void setOverlaySecurityText(String text) {
        mCameraScanModel.setOverlaySecurityText(text);
    }
}
